firebase-firestore/src/androidTest/java/com/google/firebase/firestore/QueryTest.java

```diff
@@ -1029,6 +1031,46 @@ public void testMultipleUpdatesWhileOffline() {
     assertEquals(asList(map("foo", "zzyzx", "bar", "2")), querySnapshotToValues(snapshot2));
   }
 
+  @Test
+  public void resumingQueryShouldRemoveDeletedDocumentsIndicatedByExistenceFilter() {
+    Map<String, Map<String, Object>> testDocs = new LinkedHashMap<>();
+    for (int i = 1; i <= 100; i++) {
+      testDocs.put("doc" + i, map("key", i));
+    }
+
+    // Setup firestore with disabled persistence and populate a collection with testDocs.
+    FirebaseFirestore firestore = testFirestore();
+    firestore.setFirestoreSettings(
+        new FirebaseFirestoreSettings.Builder().setPersistenceEnabled(false).build());
+    CollectionReference collection = firestore.collection(autoId());
+    writeAllDocs(collection, testDocs);
+
+    QuerySnapshot snapshot1 = waitFor(collection.get());
+    assertEquals(snapshot1.size(), 100);
+
+    // Delete 50 docs in transaction so that it doesn't affect local cache.
+    waitFor(
+        firestore.runTransaction(
+            transaction -> {
+              for (int i = 1; i <= 50; i++) {
+                DocumentReference docRef = collection.document("doc" + i);
```

Comment:
Get the names of the documents to delete from `testDocs` rather than recalculating them here.

This (a) helps with readability because it shows that you're deleting documents that you just created and (b) avoids code duplication of generating the document names twice.